ingestion_tools/scripts/data_validation/fixtures/data.py

```diff
@@ -18,24 +23,98 @@
+    fsstore_subfolders = set(fsstore.listdir())
+    expected_fsstore_subfolders = {str(i) for i in range(3)}.union({".zattrs", ".zgroup"})
+    if expected_fsstore_subfolders != fsstore_subfolders:
+        pytest.fail(f"Expected zarr subfolders: {expected_subfolders}, Actual zarr subfolders: {fsstore_subfolders}")
+
+    loc = ZarrLocation(fsstore)
+    zarrays = {}
+    for i in range(3):
+        zarrays[i] = json.loads(fsstore[str(i) + "/.zarray"].decode())
+    return {"zattrs": loc.root_attrs, "zarrays": zarrays}
 
 
-def get_headers(mrcfiles: List[str]) -> Dict[str, MrcInterpreter]:
-    """Get the mrc file headers for a list of mrc files."""
-    headers = {}
-    for file in mrcfiles:
-        try:
-            headers[file] = get_header(file)
-        except Exception as _:
-            pytest.fail(f"Failed to get header for {file}")
+# ==================================================================================================
+# Dataset fixtures
+# ==================================================================================================
 
-    return headers
+
+@pytest.fixture(scope="session")
+def dataset_metadata(dataset_metadata_file: str, filesystem: FileSystemApi) -> Dict:
+    """Load the dataset metadata."""
+    with filesystem.open(dataset_metadata_file, "r") as f:
+        return json.load(f)
+
+
+# ==================================================================================================
+# Frame fixtures
+# ==================================================================================================
+
+
+@pytest.fixture(scope="session")
+def frames_filesizes(frames_files: List[str], filesystem: FileSystemApi) -> Dict[str, int]:
+    """Get the file sizes for a list of frame files."""
+    return {frame_file: filesystem.size(frame_file) for frame_file in frames_files}
+
+
+@pytest.fixture(scope="session")
+def frames_headers(
+    frames_files: List[str],
+    filesystem: FileSystemApi,
+) -> Dict[str, Union[tifffile.TiffPages, MrcInterpreter]]:
+    """Get the headers for a list of frame files."""
+
+    def open_frame(frame_file: str):
+        if frame_file.endswith(".mrc"):
+            return (frame_file, get_header(frame_file, filesystem))
+        elif frame_file.endswith(".tif") or frame_file.endswith(".tiff") or frame_file.endswith(".eer"):
+            with filesystem.open(frame_file, "rb") as f:
+                # For some reason, just returning the tifffile.TiffFile object gives issues
+                return (frame_file, tifffile.TiffFile(f).pages)
+        else:
+            return None
+
+    # Open the images in parallel
+    with ThreadPoolExecutor() as executor:
+        headers = list(executor.map(open_frame, frames_files))
+        headers = [header for header in headers if header is not None]
+        if not headers:
+            pytest.skip("No file-format supported frames headers found")
+
+    return dict(headers)
+
```

Comment:
get all the frames' header data, multithreaded to reduce network request wait time